### PR TITLE
GEOMESA-2113 Upgrade Arrow to 0.10

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
replaces https://github.com/locationtech/geomesa/pull/1940

File: build/cqs.tsv [hunk 1]
```diff
@@ -124,9 +121,9 @@ org.antlr:antlr4-runtime	4.5	compile
 org.apache.accumulo:accumulo-core	1.8.1	compile
 org.apache.accumulo:accumulo-fate	1.8.1	compile
 org.apache.accumulo:accumulo-start	1.8.1	compile
-org.apache.arrow:arrow-format	0.6.0	compile
-org.apache.arrow:arrow-memory	0.6.0	compile
-org.apache.arrow:arrow-vector	0.6.0	compile
+org.apache.arrow:arrow-format	0.10.0	compile
+org.apache.arrow:arrow-memory	0.10.0	compile
+org.apache.arrow:arrow-vector	0.10.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17078

File: build/cqs.tsv [hunk 2]
```diff
@@ -75,12 +75,9 @@ io.airlift:aircompressor	0.8	compile
...
 io.dropwizard.metrics:metrics-graphite	3.1.2	compile
-io.netty:netty-all	4.0.41.Final	compile
-io.netty:netty-buffer	4.0.41.Final	compile
-io.netty:netty-codec	4.0.41.Final	compile
-io.netty:netty-common	4.0.41.Final	compile
-io.netty:netty-handler	4.0.41.Final	compile
-io.netty:netty-transport	4.0.41.Final	compile
+io.netty:netty-all	4.1.17.Final	compile
+io.netty:netty-buffer	4.1.17.Final	compile
+io.netty:netty-common	4.1.17.Final	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17079

**Comment by jahhulbert-ccri** [comment]
Shoudl we add the 2.1.0 milestone for this?

**Comment by elahrvivaz** [comment]
I added to the 2.1.0 milestone - looks like the current RC for arrow 0.10 is likely to be approved and released today or tomorrow.

**Comment by elahrvivaz** [comment]
this is ready to merge

File: docs/user/upgrade.rst [hunk 3]
```diff
@@ -126,8 +126,14 @@ has been replaced with standard ingest type inference, which works similarly but
 Generally, the previous behavior can be replicated by using type inference to create a converter definition,
 then modifying the converter to set the feature ID to the first column (``$1``).
 
-Scalatra Version Updates
-------------------------
+Arrow Version Update
+--------------------
+
+The version of Apache Arrow used for Arrow-encoded results has been updated from 0.6.0 to 0.10.0. Due to changes
+in the Arrow inter-process communication (IPC) format, clients may need to update to the same Arrow version.
```
Comment (jnh5y) [inline docs/user/upgrade.rst]: May?;)

Comment (elahrvivaz) [reply docs/user/upgrade.rst]: I didn't have any luck with using the java 0.6 client, but @theneuralbit said the javascript client worked fine with both...

Comment (TheNeuralBit) [reply docs/user/upgrade.rst]: Yeah there were some breaking changes to the format between 0.6 and 0.9 (not sure about 0.10), but the JS-0.3 release _should_ be able to handle it.

**Pull request merged**